File: Problemset/running-sum-of-1d-array/running-sum-of-1d-array.cpp

// @Title: 一维数组的动态和 (Running Sum of 1d Array)
// @Author: 846188037@qq.com
// @Date: 2020-07-14 21:00:06
// @Runtime: 16 ms
// @Memory: 8.4 MB

class Solution {
public:
    vector<int> runningSum(vector<int>& nums) {
        int sum=0;
        vector<int> runningSum;
        for(int i=0;i<nums.size();i++){
            for(int j=0;j<i+1;j++){
                sum+=nums[j];
            }
            runningSum.push_back(sum);
            sum=0;
        }
        return runningSum;

    }
};

File: Problemset/running-sum-of-1d-array/running-sum-of-1d-array.js

// @Title: 一维数组的动态和 (Running Sum of 1d Array)
// @Author: 846188037@qq.com
// @Date: 2020-07-14 21:15:26
// @Runtime: 100 ms
// @Memory: 34.9 MB

/**
 * @param {number[]} nums
 * @return {number[]}
 */
var runningSum = function(nums) {
    var sum=0;
    var runningSum = new Array;
    for(var i=0;i<nums.length;i++){
        for(var j=0;j<i+1;j++){
            sum+=nums[j];
        }
        runningSum.push(sum);
        sum=0;
    }
    return runningSum;

};
